catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -383,6 +384,41 @@ public static void removePermissionsFile(Project project) {
 		}
 	}
 
+	public static void updateUserDataReferences(String oldName, String newName,
+			UserData item) {
+		if (item instanceof UserList || item instanceof UserVariable) {
+			for (Scene scene : getInstance().getCurrentProject().getSceneList()) {
+				for (Sprite sprite : scene.getSpriteList()) {
+					for (Script script : sprite.getScriptList()) {
+						Brick scriptBrick = script.getScriptBrick();
```

Comment:
List<Brick> flatList = new ArrayList<Brick>();
script.addToFlatList(flatList);

for (Brick brick : flatList) {
    if (brick instanceof FormulaBrick) { .....